src/hypertable.c

```diff
@@ -1112,46 +1116,41 @@ insert_blocker_trigger_add(Oid relid)
 		.type = T_CreateTrigStmt,
 		.row = true,
 		.timing = TRIGGER_TYPE_BEFORE,
-		.trigname = INSERT_BLOCKER_NAME,	/* Note, internal triggers get the
-											 * OID appended to the name */
+		.trigname = INSERT_BLOCKER_NAME,
 		.relation = makeRangeVar(schema, relname, -1),
 		.funcname = list_make2(makeString(INTERNAL_SCHEMA_NAME), makeString(INSERT_BLOCKER_NAME)),
 		.args = NIL,
 		.events = TRIGGER_TYPE_INSERT,
 	};
 
-	objaddr.objectId = insert_blocker_trigger_get(relid);
-
-	/* Trigger already exists. Do nothing */
-	if (OidIsValid(objaddr.objectId))
-		return objaddr.objectId;
-
 	/*
-	 * Create as an internal trigger; it won't show up with \d and won't be
-	 * inherited by chunks.
+	 * We create a user-visible trigger, so that it will get pg_dump'd with
+	 * the hypertable. We overwrite any trigger that might have this name on
+	 * this table.
 	 */
-	objaddr = CreateTrigger(&stmt, NULL, relid, InvalidOid, InvalidOid, InvalidOid, true);
+	objaddr = CreateTrigger(&stmt, NULL, relid, InvalidOid, InvalidOid, InvalidOid, false);
 
 	if (!OidIsValid(objaddr.objectId))
 		elog(ERROR, "could not create insert blocker trigger");
 
 	return objaddr.objectId;
 }
 
-TS_FUNCTION_INFO_V1(hypertable_insert_blocker_trigger_add);
+TS_FUNCTION_INFO_V1(hypertable_insert_blocker_trigger_drop_and_add);
 
 /*
- * This function is exposed to add the blocking trigger on legacy hypertables
- * that don't have the trigger. We can't do it from SQL code, because internal
- * triggers cannot be added from SQL.
+ * This function is exposed to drop the old blocking trigger on legacy hypertables.
+ * We can't do it from SQL code, because internal triggers cannot be dropped from SQL.
+ * After the legacy internal trigger is dropped, we add the new, visible trigger.
  *
  * In case the hypertable's root table has data in it, we bail out with an
  * error instructing the user to fix the issue first.
  */
 Datum
-hypertable_insert_blocker_trigger_add(PG_FUNCTION_ARGS)
+hypertable_insert_blocker_trigger_drop_and_add(PG_FUNCTION_ARGS)
```

Comment:
I don't think we need to rename this function, because if there is no existing private trigger it will be equivalent to simply an `add`. This might be useful in case people remove their block triggers by accident or for some other reason and need to re-add them in a consistent way.

Reply:
changed name back to `add`